File: lev1/최소직사각형.py
def solution(sizes):
    for s in sizes:
        if s[0] < s[1]:
            t = s[0]
            s[0] = s[1]
            s[1] = t
        
    max_w, max_h = sizes[0]
    for s in sizes[1:]:
        if max_w < s[0]: max_w = s[0]
        if max_h < s[1]: max_h = s[1]
    return max_w * max_h


print(solution([[60, 50], [30, 70], [60, 30], [80, 40]]))
print(solution([[10, 7], [12, 3], [8, 15], [14, 7], [5, 15]]))
print(solution([[14, 4], [19, 6], [6, 16], [18, 7], [7, 11]]))
print(solution([[80, 10], [10, 50], [10, 60], [10, 70]]))
print(solution([[10, 70], [20, 60], [10, 50], [80, 10]]))

File: lev1/최소직사각형.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// sizes_rows는 2차원 배열 sizes의 행 길이, sizes_cols는 2차원 배열 sizes의 열 길이입니다.
int solution(int sizes[][2], size_t sizes_rows, size_t sizes_cols) {
    int h = 0;
    int w = 0;

    for(int i=0; i<sizes_rows; i++){
        if(sizes[i][0] < sizes[i][1]){
            int t = sizes[i][0];
            sizes[i][0] = sizes[i][1];
            sizes[i][1] = t;
        }
        if(h < sizes[i][0]) h = sizes[i][0];
        if(w < sizes[i][1]) w = sizes[i][1];
    }

    return h * w;
}

int main(){
    int arr[][2] = {{60, 50}, {30, 70}, {60, 30}, {80, 40}};
    printf("%d\n", solution(arr, 4, 2)); //4000
    int arr1[][2] = {{10, 7}, {12, 3}, {8, 15}, {14, 7}, {5, 15}};
    printf("%d\n", solution(arr1, 5, 2)); //120
    int arr2[][2] = {{14, 4}, {19, 6}, {6, 16}, {18, 7}, {7, 11}};
    printf("%d\n", solution(arr2, 5, 2)); //133
    return 0;
}
